### PR TITLE
docs: create docstring for all methods

### LINKED ISSUE #20: Create docstrings for all the public methods
Following the conversation on #17 , It would be great to have docstrings and document the Class and its methods.

For starters... 

- [x] which docstring syntax are we are going to stick with? (RestructuredText, Google, Numpy...) -> google

### PR DESCRIPTION
**Pull request by musantro**
I uploaded the PR in case you want to make some clarifications/modifications during the process. I will update this PR until ready, then convert from draft PR to PR ready to review.

This will close #20 

## TODO
- [x] property `cars`
- [x] method `constants_categories()`
- [x] method `constants_divisions()`
- [x] method `constants_event_types()`
- [x] method `get_carclass()`
- [x] method `get_carclasses()`
- [x] method `get_cars()`
- [x] method `get_cars_assets()`
- [x] method `get_series()`
- [x] method `get_series_assets()`
- [x] method `get_tracks()`
- [x] method `get_tracks_assets()`
- [x] method `hosted_combined_sessions()`
- [x] method `hosted_sessions()`
- [x] method `league_cust_league_sessions()`
- [x] method `league_directory()`
- [x] method `league_get()`
- [x] method `league_get_points_systems()`
- [x] method `league_membership()`
- [x] method `league_season_sessions()`
- [x] method `league_season_standings()`
- [x] method `league_seasons()`
- [x] method `lookup_club_history()`
- [x] method `lookup_countries()`
- [x] method `lookup_drivers()`
- [x] method `lookup_get()`
- [x] method `lookup_licenses()`
- [x] method `member()`
- [x] method `member_chart_data()`
- [x] method `member_info()`
- [x] method `member_profile()`
- [x] method `result()`
- [x] method `result_event_log()`
- [x] method `result_lap_chart_data()`
- [x] method `result_lap_data()`
- [x] method `result_search_hosted()`
- [x] method `result_search_series()`
- [x] method `result_season_results()`
- [x] method `season_list()`
- [x] method `season_race_guide()`
- [x] property `series`
- [x] method `series_assets()`
- [x] method `series_seasons()`
- [x] method `series_stats()`
- [x] method `stats_member_bests()`
- [x] method `stats_member_career()`
- [x] method `stats_member_recent_races()`
- [x] method `stats_member_summary()`
- [x] method `stats_member_yearly()`
- [x] method `stats_season_driver_standings()`
- [x] method `stats_season_qualify_results()`
- [x] method `stats_season_supersession_standings()`
- [x] method `stats_season_team_standings()`
- [x] method `stats_season_tt_results()`
- [x] method `stats_season_tt_standings()`
- [x] method `stats_world_records()`
- [x] method `team()`
- [x] property `tracks`

### CODE REVIEW

**Comment by jasondilworth56** [comment]
Awesome start, thanks!

**Comment by musantro** [comment]
@jasondilworth56 do you have an idea what `lookup_get()` does? I receive an empty list.

**Comment by jasondilworth56** [comment]
> @jasondilworth56 do you have any idea of what `lookup_get()` does? I receive an empty list.

I'm not sure to be honest – looks like a simple implementation of what's in `/data/docs/` just to tick it off. The docs for that endpoint are not particularly useful either.

**Comment by musantro** [comment]
> I'm not sure to be honest – looks like a simple implementation of what's in `/data/docs/` just to tick it off. The docs for that endpoint are not particularly useful either.

I'm going to leave it without doc then

**Comment by musantro** [comment]
This PR is ready to review. By creating the pull request, I've seen several things that probably I would create an issue to discuss later on, but I think this is a good start for documentation.

I have not english native language so please feel free to suggest any modifications.

File: src/iracingdataapi/client.py [hunk 1]
```diff
@@ -125,51 +125,236 @@ def series(self):
         return self._add_assets(series, series_assets, "series_id")
 
     def constants_categories(self):
+        """Fetches a list containing the racing categories.
+
+        Retrieves a list containing the racing categories and
+        the id associated to each category.
+
+        Returns:
+            A list of dicts representing each category.
```
Comment (jasondilworth56) [inline src/iracingdataapi/client.py:134]: Worth prefixing this with `list:`

File: src/iracingdataapi/client.py [hunk 2]
```diff
@@ -125,51 +125,236 @@ def series(self):
...
         return self._get_resource("/data/constants/categories")
 
     def constants_divisions(self):
+        """Fetches a list containing the racing divisions.
+
+        Retrieves a list containing the racing divisions and
+        the id associated to each division.
+
+        Returns:
+            A list of dicts representing each division.
```
Comment (jasondilworth56) [inline src/iracingdataapi/client.py:155]: Worth prefixing with `list:`

File: src/iracingdataapi/client.py [hunk 3]
```diff
@@ -125,51 +125,236 @@ def series(self):
...
+                        'relative_speed': 45,
+                        'short_name': 'Crev23'
+                    }
+                    // And +100 more...
+                ]
+        """
         return self._get_resource("/data/carclass/get")
 
     def get_carclass(self):
+        """Fetches a list containing all the car classes in the service.
```
Comment (jasondilworth56) [inline src/iracingdataapi/client.py:243]: I'm wondering whether it's worth ditching the docstring here, or replacing it with the deprecation notice?

**Review by jasondilworth56 — COMMENTED** [review]
Thank you for this, and I would never have guessed English was not your native language so no need to worry there!

One comment outside of the notes I've left in a couple of places is about examples. I think we should commit to either providing examples in every docstring, or remove them from the few that are included in this update. How big of a job would it be to add them to the ones that are missing?

File: src/iracingdataapi/client.py (hunk 1 above)
Comment (musantro) [reply src/iracingdataapi/client.py:134]: Fixed!

File: src/iracingdataapi/client.py (hunk 2 above)
Comment (musantro) [reply src/iracingdataapi/client.py:155]: Fixed!

File: src/iracingdataapi/client.py (hunk 3 above)
Comment (musantro) [reply src/iracingdataapi/client.py:243]: Yes, you're right. Fixed!

**Comment by jasondilworth56** [comment]
Yea I think that's the way to go, let's go with just removing them from everything for now for consistency.

**Comment by musantro** [comment]
Done!

**Review by jasondilworth56 — APPROVED** [review]

**Pull request merged**